File: sliding-window/minimum-size-subarray-sum.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
  public:
    int minSubArrayLen(int target, vector<int> &nums) {
      size_t start = 0, ans = SIZE_MAX, n = nums.size();

      int sum = 0;

      for (size_t end = 0; end < n; ++end) {
        sum += nums[end];

        while (sum >= target) {
          ans = min(ans, end - start + 1);
          sum -= nums[start++];
        }
      }

      return ans == SIZE_MAX ? 0 : ans;
    }
};

File: sliding-window/minimum_size_subarray_sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start, sum, min_length = 0, 0, float("inf")

        for end, value in enumerate(nums):
            sum += value

            while sum >= target:
                min_length = min(min_length, end - start + 1)

                sum -= nums[start]
                start += 1

        return 0 if min_length == float("inf") else min_length